app/src/main/java/ch/epfl/sdp/peakar/social/SocialListAdapter.java

```diff
@@ -0,0 +1,112 @@
+     * @param items list items.
+     */
+    public SocialListAdapter(Context context, int resource, List<SocialItem> items) {
+        super(context, resource, items);
+        resourceLayout = resource;
+        mContext = context;
+    }
+
+    /**
+     * Overridden method to create correct collected items.
+     * @param position position of item
+     * @param convertView item view
+     * @param parent parent view
+     * @return a view with an added item
+     */
+    public View getView(int position, View convertView, ViewGroup parent) {
+        if (convertView == null) {
+            convertView = ListAdapterInflater.createLayout(resourceLayout, mContext,parent);
+        }
+
+        setupItem(convertView, getItem(position), position + 1);
+
+        return convertView;
+    }
+
+    /**
+     * Setup the social item view, with correct info and colors.
+     * @param v social item view
+     * @param item the social item
+     * @param rank the rank the social item has
+     */
+    private void setupItem(View v, SocialItem item, int rank) {
+        if (item != null) {
+            TextView usernameText = v.findViewById(R.id.social_item_username);
+            TextView scoreText = v.findViewById(R.id.social_item_score);
+            TextView rankText = v.findViewById(R.id.social_item_rank);
+            ImageView medalView = v.findViewById(R.id.social_item_medal);
+
+            setBackgroundColor(v, item.getUid());
+            usernameText.setText(item.getUsername());
+            scoreText.setText(mContext.getResources().getString(R.string.score_display,
+                                             UIUtils.IntegerConvert(item.getScore())));
+            rankText.setText(mContext.getResources().getString(R.string.rank_display,
+                    UIUtils.IntegerConvert(rank)));
+            setMedal(medalView, rank);
+            // TODO Set profile picture, bitmap?
+        }
+    }
+
+    /**
+     * Set the correct background color
+     * @param v view to set color to.
+     * @param uid user id of social item.
+     */
+    private void setBackgroundColor(View v, String uid) {
+        int backgroundColor = R.color.White;
+
+        if (uid.equals(AuthService.getInstance().getID())) {
+            backgroundColor = R.color.LightGrey;
+        }
+
+        v.setBackgroundResource(backgroundColor);
+    }
+
+    /**
+     * Set correct medal or hide it.
+     * @param v medal view
+     * @param rank the rank of the social item.
+     */
+    private void setMedal(ImageView v, int rank) {
+        if (rank == 1) {
+            v.setImageResource(R.drawable.social_medal_gold);
+        }
+        else if (rank == 2) {
+            v.setImageResource(R.drawable.social_medal_silver);
+        }
+        else if (rank == 3) {
+            v.setImageResource(R.drawable.social_medal_bronze);
+        }
+        else {
+            v.setVisibility(View.INVISIBLE);
+        }
```

Comment:
Instead of having many if-else, should we maybe do it with a `switch` statement?